src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -445,11 +454,14 @@ void GcsActorManager::DestroyActor(const ActorID &actor_id) {
   it->second->GetMutableActorTableData()->set_timestamp(current_sys_time_ms());
   AddDestroyedActorToCache(it->second);
   const auto actor = std::move(it->second);
+
   registered_actors_.erase(it);
 
   // Clean up the client to the actor's owner, if necessary.
   if (!actor->IsDetached()) {
     RemoveActorFromOwner(actor);
+  } else {
+    runtime_env_manager_.RemoveUriReference(actor->GetActorID().Hex());
```

Comment:
Hmm I find this API a little bit weird. Why don't we just remove the URIs by name instead of storing this ID -> URI mapping? That would be more natural to me.

Reply:
Do we no longer have access to the URIs here when the actor is going out of scope? We should (it should be in the task spec)

Reply:
I agree with you that RemoveUriReference(uri) is much natural.

The reason is that we don't store it in gcs right now. It's purely memory structure. 

As for your second question, here it's reference, so when no jobs/detached actors referring it, it'll be delete.

Reply:
I think it's better to store this mapping since it makes it easier to debug when the count goes wrong. Otherwise you're just stuck with an opaque integer that's incorrect.

Reply:
Good point about debugging, SGTM